qiskit_nature/second_q/operators/majorana_op.py

```diff
@@ -0,0 +1,521 @@
+# This code is part of a Qiskit project.
+#
+# (C) Copyright IBM 2023.
+#
+# This code is licensed under the Apache License, Version 2.0. You may
+# obtain a copy of this license in the LICENSE.txt file in the root directory
+# of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+#
+# Any modifications or derivative works of this code must retain this
+# copyright notice, and modified files need to carry a notice indicating
+# that they have been altered from the originals.
+
+"""The Majorana-particle Operator."""
+
+
+from __future__ import annotations
+
+import re
+from collections import defaultdict
+from collections.abc import Collection, Mapping
+from typing import Iterator, Sequence
+
+import numpy as np
+
+from qiskit_nature.exceptions import QiskitNatureError
+
+from .polynomial_tensor import PolynomialTensor
+from .sparse_label_op import _TCoeff, SparseLabelOp, _to_number
+from .fermionic_op import FermionicOp
+
+
+class MajoranaOp(SparseLabelOp):
+    r"""N-mode Majorana operator.
+
+    A ``MajoranaOp`` represents a weighted sum of Majorana fermion operator terms.
+    These terms are encoded as sparse labels, which are strings consisting of a space-separated list
+    of expressions. Each expression must look like :code:`_<index>`, where the :code:`<index>` is a
+    non-negative integer representing the index of the mode on which the Majorana
+    creation/annihilation operator is applied. The value of :code:`index` is bound by twice the
```

Comment:
Well, I would say it still makes sense, just that the (self-adjoint) creator = annihilator in the majorana case (i.e. particles are their own antiparticles), but I omitted it to avoid confusion. You're right that it's not really helpful for understanding the notation.